packages/effect/src/internal/stream.ts

```diff
@@ -6194,6 +6194,23 @@ export const tap: {
   ): Stream.Stream<A, E | E2, R | R2> => mapEffectSequential(self, (a) => Effect.as(f(a), a))
 )
 
+/** @internal */
+export const tapStart: {
```

Comment:
I think `onStart` is a better name, aligns with `onInterrupt` etc.

Reply:
Or maybe `onSubscribe`? 😃

Reply:
But yeah, `on` prefix sounds better, because `tap` implies some input value

Reply:
We don't use the `subscribe` terminology in effect, so I prefer "start".